### PR TITLE
autoconf: Update autoconf image file to use Ubuntu distro

### PR DESCRIPTION
**Pull request by corona10**
- Port autoconf image from alpine distro to Ubuntu distro.

### CODE REVIEW

File: autoconf/Dockerfile [hunk 1]
```diff
@@ -10,37 +10,33 @@ LABEL org.opencontainers.image.authors="Donghee Na"
...
-        pkgconfig \
-        xz
+RUN apt-get update && \
+    apt-get -yq install \
+    autotools-dev \
+    autoconf \
+    autoconf-archive \
+    build-essential \
+    curl \
+    pkg-config
```
Comment (erlend-aasland) [inline autoconf/Dockerfile:20]: Nit; I'd prefer to keep the extra indent, to make it obvious that these are arguments to `apt-get` and not a new chain of commands :)

File: autoconf/Dockerfile [hunk 2]
```diff
@@ -10,37 +10,33 @@ LABEL org.opencontainers.image.authors="Donghee Na"
...
 
-RUN set -o pipefail \
-    && curl https://ftp.gnu.org/gnu/autoconf/autoconf-${AUTOCONF_VERSION}.tar.gz | tar -zxf - \
+RUN curl https://ftp.gnu.org/gnu/autoconf/autoconf-${AUTOCONF_VERSION}.tar.gz | tar -zxf - \
```
Comment (erlend-aasland) [inline autoconf/Dockerfile:22]: Does it not work with `pipefail`?

Comment (corona10) [reply autoconf/Dockerfile:22]: When I tested, yes.
https://askubuntu.com/questions/886537/set-e-o-pipefail-not-working-on-bash-script-on-ubuntu-16

**Comment by corona10** [comment]
Hmm https://ftp.gnu.org/gnu/autoconf/autoconf-2.69.tar.gz is unstable.

**Comment by corona10** [comment]
@erlend-aasland PTAL CI failure
```
https://github.com/python/cpython-devcontainers/actions/runs/11136086934/job/30947270766?pr=13
```

File: autoconf/Dockerfile (hunk 2 above)
Comment (corona10) [reply autoconf/Dockerfile:23]: Done: https://github.com/python/cpython-devcontainers/pull/13/commits/b200e7a1668d340a8ffcc30f01307d565a976584

**Comment by corona10** [comment]
@erlend-aasland gentle ping~

File: autoconf/Dockerfile (hunk 2 above)
Comment (erlend-aasland) [reply autoconf/Dockerfile:23]: > Done: https://github.com/python/cpython-devcontainers/pull/13/commits/b200e7a1668d340a8ffcc30f01307d565a976584

Elegant!

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**